ta/pkcs11/src/processing_asymm.c

```diff
@@ -903,3 +933,159 @@ enum pkcs11_rc do_asymm_derivation(struct pkcs11_session *session,
 
 	return rc;
 }
+
+static enum pkcs11_rc wrap_rsa_aes_key(struct active_processing *proc,
+				       void *data, uint32_t data_sz,
+				       void *out_buf, uint32_t *out_sz)
+{
+	TEE_Result res = TEE_ERROR_GENERIC;
+	TEE_Attribute tee_attrs[2] = { };
+	struct rsa_aes_key_wrap_processing_ctx *ctx = proc->extra_ctx;
+	mbedtls_nist_kw_context kw_ctx;
+	uint8_t aes_key_value[32] = { };
+	uint32_t aes_key_size = ctx->aes_key_bits / 8;
+	uint32_t tee_attrs_count = 0;
+	uint32_t aes_wraped_size = *out_sz;
+	size_t target_key_size = 0;
+
+	if (ctx->aes_key_bits != 128 &&
+	    ctx->aes_key_bits != 192 &&
+	    ctx->aes_key_bits != 256)
+		return PKCS11_CKR_ARGUMENTS_BAD;
+
+	mbedtls_nist_kw_init(&kw_ctx);
+	TEE_GenerateRandom(aes_key_value, aes_key_size);
+	res = TEE_AsymmetricEncrypt(proc->tee_op_handle,
+				    tee_attrs, tee_attrs_count,
+				    aes_key_value, aes_key_size,
+				    out_buf, &aes_wraped_size);
+	if (res) {
+		res = tee2pkcs_error(res);
+		goto out;
+	}
+
+	res = mbedtls_nist_kw_setkey(&kw_ctx, MBEDTLS_CIPHER_ID_AES,
```

Comment:
Here and at line 974: use a specific `int mbedtls_rc` or like for mdebtls_xxx() return value.
At line 970 (and 979), set `rc` (see comment above).
Or straight:
```c
	if (mbedtls_nist_kw_setkey(...)) {
		rc = PKCS11_CKR_ARGUMENTS_BAD;
		goto out;
	}
```

ditto in `unwrap_rsa_aes_key()`

Reply:
done